### ISSUE TITLE
FileContext.sourceFile path on Windows has invalid uri.path for saving

### ISSUE THREAD

**Issue by TimWhiting** [Workiva/dart_codemod#44]
See https://github.com/rrousselGit/river_pod/issues/422 for full issue report. Stacktrace below.

```sh
Unhandled exception:
FileSystemException: Cannot open file, path = '/e:/workspace/myproject/lib/provider/client_detail_provider.dart' (OS Error: El nombre de archivo, el nombre de directorio o la sintaxis de la etiqueta del volumen no son correctos., errno = 123)
#0      _File.throwIfError (dart:io/file_impl.dart:635:7)
#1      _File.openSync (dart:io/file_impl.dart:479:5)
#2      _File.writeAsBytesSync (dart:io/file_impl.dart:604:31)
#3      _File.writeAsStringSync (dart:io/file_impl.dart:628:5)
#4      applyPatchesAndSave (package:codemod/src/util.dart:79:29)
#5      _runInteractiveCodemod (package:codemod/src/run_interactive_codemod.dart:314:9)
<asynchronous suspension>
```

**Comment by evanweible-wf** [comment]
@TimWhiting Thanks for the report! I don't have access to a windows machine, but based on the stack trace I _think_ this would fix it:
https://github.com/Workiva/dart_codemod/pull/45

Would you mind creating a branch of river_pod that uses that branch and asking the original issue author if they can try it out?

**Comment by TimWhiting** [comment]
Yes, this fixes the issue. They were able to migrate successfully after this.

**Comment by evanweible-wf** [comment]
Fix has been published in 0.3.1
